packages/braintree-integration/src/braintree-paypal-ach/braintree-paypal-ach-payment-strategy.ts

```diff
@@ -69,7 +74,103 @@ export default class BraintreePaypalAchPaymentStrategy implements PaymentStrateg
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
-        const { paymentData } = payment;
+        if (!this.paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        try {
+            const nonce = await this.tokenizePayment(payment);
+            const submitPaymentPayload = await this.preparePaymentData(nonce, payment);
+
+            await this.paymentIntegrationService.submitOrder(order, options);
+            await this.paymentIntegrationService.submitPayment(submitPaymentPayload);
+        } catch (error) {
+            this.handleError(error);
+        }
+    }
+
+    finalize(): Promise<void> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    async deinitialize(): Promise<void> {
+        this.getMandateText = undefined;
+
+        return Promise.resolve();
+    }
+
+    private async preparePaymentData(
+        nonce: string | null,
+        payment: OrderPaymentRequestBody,
+    ): Promise<Payment> {
+        const { paymentData = {} } = payment;
+
+        if (!nonce) {
+            if (!isVaultedInstrument(paymentData)) {
+                throw new PaymentArgumentInvalidError(['payment.paymentData']);
+            }
+
+            return {
+                methodId: payment.methodId,
+                paymentData: {
+                    instrumentId: paymentData.instrumentId,
+                },
+            };
+        }
+
+        if (!isUsBankAccountInstrumentLike(paymentData)) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        const sessionId = await this.braintreeIntegrationService.getSessionId();
+
+        const state = this.paymentIntegrationService.getState();
+
+        const { email } = state.getCheckoutOrThrow().billingAddress || { email: null };
+
+        const { shouldSaveInstrument, shouldSetAsDefaultInstrument, routingNumber, accountNumber } =
+            paymentData;
+
+        const paymentPayload = {
+            formattedPayload: {
+                vault_payment_instrument: shouldSaveInstrument || null,
+                set_as_default_stored_instrument: shouldSetAsDefaultInstrument || null,
+                device_info: sessionId || null,
+                ach_account: {
+                    routing_number: routingNumber,
+                    last4: accountNumber.substr(-4),
+                    token: nonce,
+                    email: email || null,
+                },
+            },
+        };
```

Comment:
Is it possible to use billing address from state rather from checkout object?

```suggestion
        const billingAddress = state.getBillingAddressOrThrow();

        const { shouldSaveInstrument, shouldSetAsDefaultInstrument, routingNumber, accountNumber } =
            paymentData;

        const paymentPayload = {
            formattedPayload: {
                vault_payment_instrument: shouldSaveInstrument || null,
                set_as_default_stored_instrument: shouldSetAsDefaultInstrument || null,
                device_info: sessionId || null,
                ach_account: {
                    routing_number: routingNumber,
                    last4: accountNumber.substr(-4),
                    token: nonce,
                    email: billingAddress.email || null,
                },
            },
        };
```